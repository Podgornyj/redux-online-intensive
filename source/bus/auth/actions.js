import { types } from "./types";

export const authActions = {
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        };
    },
    authenticateAsync: () => {
        return {
            type: types.AUTHENTICATE_ASYNC,
        };
    },
    signupAsync: (userData) => {
        return {
            type: types.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    loginAsync: (userData) => {
        return {
            type: types.LOGIN_ASYNC,
            payload: userData,
        };
    },
    initialize: () => {
        return {
            type: types.INITIALIZE,
        };
    },
    initializeAsync: () => {
        return {
            type: types.INITIALIZE_ASYNC,
        };
    },
    logout: () => {
        return {
            type: types.LOGOUT,
        };
    },
    logoutAsync: () => {
        return {
            type: types.LOGOUT_ASYNC,
        };
    },
};
