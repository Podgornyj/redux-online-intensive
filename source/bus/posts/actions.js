import { types } from './types';
import { api } from '../../REST';


export const  postsActions = {
    fillPosts : (posts) => {
        return {
            type: types.FILL_POSTS,
            payload: posts,
        };
    },
    fetchPostsAsync : () => {
        return {
            type: types.FETCH_POST_ASYNC,
        };
    },
    fillNewPost : (post) => {
        return {
            type: types.FILL_NEW_POST,
            payload: post,
        };
    },
    createPostAsync : (comment) => {
        return {
            type: types.CREATE_POST_ASYNC,
            payload: comment,
        };
    },
};
