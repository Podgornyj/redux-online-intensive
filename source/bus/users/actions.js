import { types } from "./types";

export const userActions = {
    clearUsers: () => {
        return {
            type: types.CLEAR_USERS,
        };
    },
    fillUsers: (users) => {
        return {
            type: types.FILL_USERS,
            payload: users,
        };
    },
    fetchUsersAsync: () => {
        return {
            type: types.FETCH_USERS_ASYNC,
        };
    },
};
