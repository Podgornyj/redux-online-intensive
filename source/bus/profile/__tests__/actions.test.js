import { profileActions } from '../actions';

test('Test fillProfile action', () => {
    expect(profileActions.fillProfile(__.userProfile)).toMatchSnapshot();
});
test('Test updateAvatar action', () => {
    expect(profileActions.updateAvatar(__.url)).toMatchSnapshot();
});
test('Test clearProfile action', () => {
    expect(profileActions.clearProfile()).toMatchSnapshot();
});
test('Test updateNameAsync action', () => {
    expect(profileActions.updateNameAsync(__.newName)).toMatchSnapshot();
});
test('Test updateAvatarAsync action', () => {
    expect(profileActions.updateAvatarAsync(__.newAvatar)).toMatchSnapshot();
});
test('Test updatePasswordAsync action', () => {
    expect(profileActions.updatePasswordAsync(__.newPassword)).toMatchSnapshot();
});
