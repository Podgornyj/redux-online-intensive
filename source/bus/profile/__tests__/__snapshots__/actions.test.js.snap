// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test clearProfile action 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`Test fillProfile action 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE ",
}
`;

exports[`Test updateAvatar action 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;

exports[`Test updateAvatarAsync action 1`] = `
Object {
  "payload": Array [
    "avatar",
  ],
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`Test updateNameAsync action 1`] = `
Object {
  "payload": Object {
    "firstName": "Anton",
    "lastName": "Podgornyj",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`Test updatePasswordAsync action 1`] = `
Object {
  "payload": Object {
    "newPassword": "123456",
    "oldPassword": "12345",
  },
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
